p8e-util/src/main/kotlin/io/p8e/crypto/SmartKeySigner.kt

```diff
@@ -44,10 +45,14 @@ import java.security.Signature
 class SmartKeySigner(
     private val signAndVerifyApi: SignAndVerifyApi,
     private val securityObjectsApi: SecurityObjectsApi,
+    private val authenticationApi: AuthenticationApi
 ): SignerImpl {
 
     init {
         Security.addProvider(BouncyCastleProvider())
+
+        // Kick background thread to refresh SmartKey's auth api.
+        SmartKeyAuthRefresh(authenticationApi).refreshSmartKeyAuthToken()
```

Comment:
Refreshing token at the signer, cause before we can encrypt we would need to sign the data.